shotgun_api3/shotgun.py

```diff
@@ -371,6 +375,21 @@ def __init__(self):
         self.authorization = None
         self.no_ssl_validation = False
 
+    @property
+    def records_per_page(self):
+        """
+        The records per page value from the server.
+        """
+        if self._records_per_page is None:
+            # Check for api_max_entities_per_page in the server info and change the record per page
+            # value if it is supplied.
+            self._records_per_page = self._sg.server_info.get('api_max_entities_per_page') or 500
+        return self._records_per_page
+
+    @records_per_page.setter
+    def records_per_page(self, records_per_page):
+        self._records_per_page = records_per_page
```

Comment:
Should we make sure someone doesn't set the value higher than `api_max_entities_per_page`?